mypy_django_plugin/transformers/managers.py

```diff
@@ -115,6 +117,48 @@ def get_funcdef_type(definition: Union[FuncBase, Decorator, None]) -> Optional[P
     )
 
 
+def _manage_type_var(ret_type: MypyType, queryset_info: TypeInfo) -> MypyType:
+    """
+    Attempts to manage the concrete type corresponding to a `TypeVarType` in a queryset's base types.
+    If nothing is found, keep it unchanged.
+
+    Example:
+
+    T = TypeVar("T")
+    for def example_list_dict(self) -> list[dict[str, T]]
+
+    Tries to find the corresponding `TypeVar` of `T` if it has been set.
+
+    """
+    if isinstance(ret_type, TypeVarType):
+        return _find_type_var(queryset_info, ret_type) or ret_type
+    elif isinstance(ret_type, Instance):
+        # Since it is an instance, recursively find the type var for all its args.
+        ret_type.args = tuple(_manage_type_var(item, queryset_info) for item in ret_type.args)
+    if isinstance(ret_type, ProperType) and hasattr(ret_type, "item"):
+        # For example TypeType has an item. find the type_var for this item
+        ret_type.item = _manage_type_var(ret_type.item, queryset_info)
+    if isinstance(ret_type, ProperType) and hasattr(ret_type, "items"):
+        # For example TypeList has items. find recursively type_var for its items
+        ret_type.items = [_manage_type_var(item, queryset_info) for item in ret_type.items]
+    return ret_type
+
+
+def _find_type_var(queryset_info: TypeInfo, ret_type: TypeVarType) -> Optional[MypyType]:
+    """
+    Attempts to find the concrete type corresponding to a TypeVarType in a queryset's base types.
+
+    Example:
+    Suppose queryset_info is based on a generic class `QuerySet[T]` and ret_type corresponds
+    to `T`, then this function will return the concrete type that `T` was instantiated with.
+    """
+    for base in queryset_info.bases:
```

Comment:
Hm, using `TypeInfo.bases` would only include direct base classes while also not necessarily honoring the mro, I think.

I think a breaking case then is if you put half of your custom methods in your test in another parent class, under a `TypeVar` named differently, it won't be found. e.g.

```python
class _MyModelQuerySet(models.QuerySet[_CTE]):
    def example(self) -> _CTE: ...

class _MyModelQuerySet2(_MyModelQuerySet[_CTE2]):
    def example_2(self) -> _CTE_2: ...

class MyModelQuerySet(_MyModelQuerySet2["MyModel"]):
    ...

...

reveal_type(MyModel.objects.example())
```